### PR TITLE
Add workflow for workitem validation

### PR DESCRIPTION
**Pull request by aholstrup1**
# Summary 
This PR adds some workitem validation to PRs coming from **forks**. Workitem validation for PRs coming from branches will come in the next iteration. On every PR from a fork we want to ensure that: 
* There is a linked issue 
* The linked issue exists, is open and is approved 

If those criteria are not met, the workitem validation will fail and we post a comment to the PR. 

Additionally, we will also add the pull request to a GitHub milestone 

# Work Item(s) 
Fixes #

### CODE REVIEW

File: .github/workflows/WorkitemValidation.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: Work Item Validation
+on:
+  pull_request_target:
```
Comment (mazhelez) [inline .github/workflows/WorkitemValidation.yaml:3]: Let's explicitly specify the types on which the workflow is triggered.

From https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target

What about the following:
- opened
- edited
- reopened
- synchronize

File: build/scripts/GitHub/GitHubIssue.class.psm1 [hunk 2]
```diff
@@ -0,0 +1,65 @@
+using module .\GitHubAPI.class.psm1
+
+
```
Comment (mazhelez) [inline build/scripts/GitHub/GitHubIssue.class.psm1]:
```suggestion
```

File: build/scripts/GitHub/GitHubPullRequest.class.psm1 [hunk 3]
```diff
@@ -0,0 +1,105 @@
+using module .\GitHubAPI.class.psm1
+using module .\GitHubIssue.class.psm1
+
+
```
Comment (mazhelez) [inline build/scripts/GitHub/GitHubPullRequest.class.psm1]:
```suggestion
```

File: build/scripts/GitHub/GitHubPullRequest.class.psm1 [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    [int] $PRNumber
+    [string] $Repository
+    $PullRequest
+
+    hidden GitHubPullRequest([int] $PRNumber, [string] $Repository) {
+        $this.PRNumber = $PRNumber
+        $this.Repository = $Repository
+
+        $pr = gh api "/repos/$Repository/pulls/$PRNumber" -H ([GitHubAPI]::AcceptJsonHeader) -H ([GitHubAPI]::GitHubAPIHeader) | ConvertFrom-Json
+        if ($pr.message) {
```
Comment (mazhelez) [inline build/scripts/GitHub/GitHubPullRequest.class.psm1:17]: Should the logic be the same as in `GitHubIssue`?

File: build/scripts/PullRequestValidation/ValidateIssuesForPullRequest.ps1 [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        $pullRequest.AddComment($Comment)
+        $invalidIssues += $issueId
+    }
+    else {
+        $pullRequest.RemoveComment($Comment)
+    }
+}
+
+if($invalidIssues) {
+    throw "The following issues are not open or approved: $($invalidIssues -join ', ')"
```
Comment (mazhelez) [inline build/scripts/PullRequestValidation/ValidateIssuesForPullRequest.ps1]:
```suggestion
    throw "The following issues are not valid: $($invalidIssues -join ', '). Issues linked in a PR need to exist, be open and approved."
```

File: build/scripts/PullRequestValidation/ValidateIssuesForPullRequest.ps1 [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+$pullRequest.RemoveComment($Comment)
+
+$invalidIssues = @()
+
+foreach ($issueId in $issueIds) {
+    Write-Host "Validating issue $issueId"
+    $issue = [GitHubIssue]::Get($issueId, $Repository)
+
+    # If the issue is not approved, add a comment to the pull request and throw an error
```
Comment (mazhelez) [inline build/scripts/PullRequestValidation/ValidateIssuesForPullRequest.ps1]:
```suggestion
    # Add a comment to the PR for each invalid issue.
```

**Review by mazhelez — APPROVED** [review]

**Review by elbardel — APPROVED** [review]

**Pull request merged**